code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -427,6 +432,13 @@
 	extract_major = new(src)
 	extract_major.Grant(C)
 
+/datum/species/jelly/luminescent/spec_life(mob/living/carbon/human/H)
+	if(extract_minor)
+		extract_minor.UpdateButtonIcon(TRUE) //status_only
+	if(extract_major)
+		extract_major.UpdateButtonIcon(TRUE) //status_only
+	..()

```

Comment:
These icons used to update at a constant rate and it turned out they were causing lag, so it got removed. I don't know if this would make that problem reappear, but if possible make a performance test before this is merged.

Reply:
As i see here
https://github.com/tgstation/tgstation/blob/dfecf744efe651e98f7af40ab4945b74368b995e/code/datums/action.dm#L115-L138
`status_only` update only color.
So this should not cause lags.

Reply:
ah, you're right, i did initially place it inside IsAvailable

Reply:
Some profile screenshots.
<details>
  <summary>Spoiler</summary>

![image](https://user-images.githubusercontent.com/7734424/75273131-1b58ab80-5808-11ea-91f5-88fb572114e5.png)

![image](https://user-images.githubusercontent.com/7734424/75273157-34f9f300-5808-11ea-9331-e25f63b5b2d2.png)

<details>

I don't completely understand on what need see here.

Reply:
Why is this even necessary from quick glance these shouldn't be changing randomly ?

Reply:
if i remember correctly, action cooldown is handled by a counter, not a timer, so the only way to detect when the cooldown is over is by checking periodically

Reply:
Changing that would be better idea than this then.